homeassistant/components/sql/sensor.py

```diff
@@ -58,67 +51,101 @@ def validate_sql_select(value: str) -> str:
+    )
 
-    sess: scoped_session | None = None
-    try:
-        engine = sqlalchemy.create_engine(db_url)
-        sessmaker = scoped_session(sessionmaker(bind=engine))
-
-        # Run a dummy query just to test the db_url
-        sess = sessmaker()
-        sess.execute("SELECT 1;")
-
-    except sqlalchemy.exc.SQLAlchemyError as err:
-        _LOGGER.error(
-            "Couldn't connect using %s DB_URL: %s",
-            redact_credentials(db_url),
-            redact_credentials(str(err)),
+    for query in config[CONF_QUERIES]:
+        new_config = {
+            CONF_DB_URL: config.get(CONF_DB_URL, default_db_url),
+            CONF_NAME: query.get(CONF_NAME),
+            CONF_QUERY: query.get(CONF_QUERY),
+            CONF_UNIT_OF_MEASUREMENT: query.get(CONF_UNIT_OF_MEASUREMENT),
+            CONF_VALUE_TEMPLATE: query.get(CONF_VALUE_TEMPLATE),
+            CONF_COLUMN_NAME: query.get(CONF_COLUMN_NAME),
+        }
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=new_config,
+            )
         )
-        return
-    finally:
-        if sess:
-            sess.close()
 
-    queries = []
 
-    for query in config[CONF_QUERIES]:
-        name: str = query[CONF_NAME]
-        query_str: str = query[CONF_QUERY]
-        unit: str | None = query.get(CONF_UNIT_OF_MEASUREMENT)
-        value_template: Template | None = query.get(CONF_VALUE_TEMPLATE)
-        column_name: str = query[CONF_COLUMN_NAME]
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the SQL sensor entry."""
+
+    db_url: str = entry.options[CONF_DB_URL]
+    name: str = entry.options[CONF_NAME]
+    query_str: str = entry.options[CONF_QUERY]
+    unit: str | None = entry.options.get(CONF_UNIT_OF_MEASUREMENT)
+    template: str | None = entry.options.get(CONF_VALUE_TEMPLATE)
+    column_name: str = entry.options[CONF_COLUMN_NAME]
 
+    value_template: Template | None = None
+    if template is not None:
+        try:
+            value_template = Template(template)
+            value_template.ensure_valid()
+        except TemplateError:
+            value_template = None
         if value_template is not None:
             value_template.hass = hass
 
-        # MSSQL uses TOP and not LIMIT
-        if not ("LIMIT" in query_str.upper() or "SELECT TOP" in query_str.upper()):
-            query_str = (
-                query_str.replace("SELECT", "SELECT TOP 1")
-                if "mssql" in db_url
-                else query_str.replace(";", " LIMIT 1;")
-            )
+    try:
+        engine = sqlalchemy.create_engine(db_url)
+        sessmaker = scoped_session(sessionmaker(bind=engine))
+    except SQLAlchemyError as err:
+        _LOGGER.error("Can not open database %s", {redact_credentials(str(err))})
+        return
```

Comment:
In the future, we should set `future=True` for this integration as its the only thing left in the codebase not using the 2.0 api

https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#migration-to-2-0-step-four-use-the-future-flag-on-engine